### PR TITLE
Create textdecoder-fatal-single-byte.html

### PR DESCRIPTION
**Pull request by ArkadiuszMichalski**
I see that Chrome doesn't throw for undefined pointer when fatal flag is set so some test would be useful (https://github.com/w3c/web-platform-tests/issues/2892).

### CODE REVIEW

File: encoding/textdecoder-fatal-single-byte.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+     {encoding: 'windows-1253', bad: [0xAA, 0xD2, 0xFF]},
+     {encoding: 'windows-1254', bad: []},
+     {encoding: 'windows-1255', bad: [0xCA, 0xD9, 0xDA, 0xDB, 0xDC, 0xDD, 0xDE, 0xDF, 0xFB, 0xFC, 0xFF]},
+     {encoding: 'windows-1256', bad: []},
+     {encoding: 'windows-1257', bad: [0xA1, 0xA5]},
+     {encoding: 'windows-1258', bad: []},
+     {encoding: 'x-mac-cyrillic', bad: []},
+];
+
+singleByteEncodings.forEach(function(t) {
```
Comment (inexorabletash) [inline encoding/textdecoder-fatal-single-byte.html]: Try and reserve the identifier `t` for a reference to a test instance, e.g. `test(function(t) { ... })`

File: encoding/textdecoder-fatal-single-byte.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    Array.apply(null, Array(256)).forEach(function(a,i) {
```
Comment (inexorabletash) [inline encoding/textdecoder-fatal-single-byte.html]: Why not just `for (var i = 0; i <256; ++i) { ... }`

File: encoding/textdecoder-fatal-single-byte.html [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        if (t.bad.length && t.bad.indexOf(i) != -1) {
```
Comment (inexorabletash) [inline encoding/textdecoder-fatal-single-byte.html]: the `t.bad.length` test is redundant.

File: encoding/textdecoder-fatal-single-byte.html (hunk 1 above)
Comment (ArkadiuszMichalski) [inline encoding/textdecoder-fatal-single-byte.html]: So I should use other name for that parameters? Firstly I reading some examples from http://w3c-test.org/encoding/ and most of them use that style.

Comment (inexorabletash) [inline encoding/textdecoder-fatal-single-byte.html]: Not a big deal. And those other tests are probably by me. You can leave it as-is.

**Comment by inexorabletash** [comment]
Okay, looks good to me. Thanks for putting this together!

**Pull request merged**